mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala

```diff
@@ -562,4 +562,20 @@ trait HasBlockSize extends Params {
   /** @group expertGetParam */
   final def getBlockSize: Int = $(blockSize)
 }
+
+/**
+ * Trait for shared param k. This trait may be changed or
+ * removed between minor versions.
+ */
+trait HasK extends Params {
+
+  /**
+   * Param for The number of clusters to create. Must be (&gt; 1). Note that it is possible for fewer than k clusters to be returned.
+   * @group param
+   */
+  final val k: IntParam = new IntParam(this, "k", "The number of clusters to create. Must be (> 1). Note that it is possible for fewer than k clusters to be returned", ParamValidators.gt(1))
+
+  /** @group getParam */
+  final def getK: Int = $(k)
```

Comment:
If this isn't final, do you avoid the MiMa problems? they weren't final before. I don't see why someone would override them, but, I suppose we can just not change that.

Reply:
Right, no MiMia problems if this isn't final, but we can't change it because it's generated code.

Reply:
OK. I guess I'm neutral on the change then, and it's really minor either way. @zhengruifeng do you have a perspective?

Reply:
I am also neutral on this change.